container-core/src/main/java/com/yahoo/container/di/CloudSubscriberFactory.java

```diff
@@ -24,11 +29,25 @@ public class CloudSubscriberFactory implements SubscriberFactory {
 
     private final ConfigSource configSource;
     private final Map<CloudSubscriber, Integer> activeSubscribers = new WeakHashMap<>();
+    private final ExecutorService executor;
 
     private Optional<Long> testGeneration = Optional.empty();
 
     public CloudSubscriberFactory(ConfigSource configSource) {
         this.configSource = configSource;
+        executor = new ThreadPoolExecutor(1, Integer.MAX_VALUE,
+                1, TimeUnit.SECONDS, new SynchronousQueue<>(),
+                new DaemonThreadFactory("cloud-subscriber-factory"));
+    }
+
+    @Override
+    public void close() {
+        executor.shutdown();
+        try {
+            if ( ! executor.awaitTermination(10, TimeUnit.SECONDS)) {
+                executor.shutdownNow();
+            }
+        } catch (InterruptedException e) { }
```

Comment:
Would it be appropriate to log something here?

Reply:
Yes, I will investigate a bit more if I can unwarp on ExecutionException or log.